pkg/controller/virtualmachine/virtualmachine_controller.go

```diff
@@ -123,12 +124,22 @@ func (r *ReconcilePolicy) addFinalizerAndUpdate(virtualMachine *kubevirt.Virtual
 	}
 
 	logger.V(1).Info("The VM does not have a finalizer")
-	virtualMachine.ObjectMeta.Finalizers = append(virtualMachine.ObjectMeta.Finalizers, pool_manager.RuntimeObjectFinalizerName)
 
-	err := r.Update(context.Background(), virtualMachine)
-	if err != nil {
-		logger.Error(err, "failed to update the VM with the new finalizer")
+	err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		// refresh vm instance
+		err := r.Get(context.TODO(), request.NamespacedName, virtualMachine)
```

Comment:
In the original code we for some reason pass virtualMachine down to this function. With your change, you overwrite it with whatever is found on the cluster. Aren't we losing some value? If so, we need to change this. If we don't, we should pass only name and namespace to make this clearly unrelated.

Reply:
This function is call if we can get a virtual machine at `Reconcile` so only virtualmachines are reaching this function.

Reply:
we don't loose values, so I should pass only name & namespace like you suggest. 
But we do use the vm instance before the retryOnConflict to check if we already have a finalizer, but that could also be done outside of this function (and it will still be ok since it's name is `addFinalizerAndUpdate`.